server/sonar-server/src/main/java/org/sonar/server/permission/PermissionService.java

```diff
@@ -76,6 +77,11 @@ public PermissionService(DbClient dbClient, PermissionRepository permissionRepos
     return GlobalPermissions.ALL;
   }
 
+  public UserWithPermissionQueryResult findUsersWithPermission(PermissionQuery query) {
```

Comment:
moreover it makes sense to have permissions checked at the service level. Why not ? We don't need to check permissions at the WS level